### DISCUSSION TITLE
TypeError: this.namespace.methods is undefined

### DISCUSSION THREAD

**Discussion by 154286** [WalletConnect/walletconnect-monorepo#3382] [Developer Support] [answered]
**Describe the bug**

When I use `modal.connect` to connect the wallet, it generates a session and saves it in local storage, which is normal. However, when I use `EthereumProvider.init` to generate an ETHProvider, the ETHProvider can access the session from the previous `modal.connect` and does not need to reconnect. But when I call `await ETHProvider.request({ method: "eth_sendTransaction", params: [tx], })`, it throws an error "TypeError: this.namespace.methods is undefined."

In this situation, I can only use `ETHProvider.disconnect()` to disconnect, and then use `ETHProvider.connect` to create a new connection. After that, when I call `ETHProvider.request({ method: "eth_sendTransaction" })`, it works without errors, and I can successfully initiate the transaction.

**SDK Version (if relevant)**
 - Client: [JS]
 - Version [
 -  "@walletconnect/modal-sign-html": "2.6.0",
 -  "@walletconnect/ethereum-provider": "2.9.1"
 - ]

**To Reproduce**
1. new WalletConnectModalSign().connect()
2. await EthereumProvider.init().request({ method: "eth_sendTransaction"})
4. See error TypeError: this.namespace.methods is undefined

**Expected behavior**
I want to know what's different between the session generated by `modal.connect` and the session generated by `ETHProvider.connect`, and why the former throws an error while the latter doesn't.

**Screenshots**

**Desktop (please complete the following information):**
 
 - Browser [chrome, edge,firefox]
 - Version [ Any version will ]

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone13]
 - OS: [e.g. iOS15.7]
 - Version [Any version will]

**Additional context**
I initially thought it was a problem with the `modal.connect` parameters. However, even when I added `eth_sendTransaction` to the `methods` parameter, it didn't resolve the issue. Below is the code for `modal.connect`:
```
  const modal = new WalletConnectModalSign({
      projectId: <myProjectID>,
      metadata: { },
    })

    const chains = ['eip155:5',];

  const session = await modal.connect({
            requiredNamespaces: {
              eip155: {
                methods: ['eth_sendTransaction', 'personal_sign', "wallet_switchEthereumChain", "eth_chainId"],
                chains,
                events: ['chainChanged', 'accountsChanged']
              }
            }
          })
```
ETHProvider:
```
const ETHProvider = await EthereumProvider.init({
    projectId: <myProjectID>, 
    chains: [5], 
    methods: [
      "eth_sendTransaction",
      "personal_sign",
      "wallet_switchEthereumChain",
      "eth_chainId",
      "eth_requestAccounts",
    ], 
   showQrModal: true
```

**Comment by glitch-txs** [comment]
hey seems like you're trying to use two different packages, could you provide a minimal reproducible example of you code?
Also feel free to check our [examples repo](https://github.com/WalletConnect/web3modal-examples) and [docs](https://docs.walletconnect.com/2.0/web/providers/ethereum)

**Comment by 154286** [comment]
I later gave up using `modal-sign-html` and only used `EthereumProvider` to avoid this problem.
But when used simultaneously, errors will still be reported

**Reply by glitch-txs** [reply]
you shouldn't be using them simultaneously, this is currently not supported

**Reply by 154286** [reply, marked as answer]
Yes, only EthereumProvider has been used, and everything is currently working properly. Should this issue be closed?

**Reply by glitch-txs** [reply]
Feel free to close 👍
